src/coreclr/tools/Common/TypeSystem/IL/ILStackHelper.cs

```diff
@@ -25,16 +25,16 @@ public static void CheckStackBalance(this MethodIL methodIL)
         public static int ComputeMaxStack(this MethodIL methodIL)
         {
             const int StackHeightNotSet = int.MinValue;
+            const int StackAllocThreshold = 256 / sizeof(int);
 
-            byte[] ilbytes = methodIL.GetILBytes();
-            int currentOffset = 0;
+            var ilReader = new ILReader(methodIL.GetILBytes());
             int stackHeight = 0;
             int maxStack = 0;
 
-            // TODO: Use Span<T> for this and stackalloc the array if reasonably sized
-            int[] stackHeights = new int[ilbytes.Length];
-            for (int i = 0; i < stackHeights.Length; i++)
-                stackHeights[i] = StackHeightNotSet;
+            Span<int> stackHeights = ilReader.Size <= StackAllocThreshold ?
```

Comment:
Could we get away with `short` or `ushort` representing the stack height 🤔 Quite lot wasted bytes here..

Reply:
I think we could. The ILAsm syntax documents this as Int32, but the file format can only express 2 bytes of MaxStack for each method body.

Reply:
Hmm, [let's try it(?)](https://github.com/dotnet/runtime/pull/97794/commits/f40fcacfb2acdcaa880ad4988fc12c0f0ae21429) and hope `short.MaxValue` is enough stack for everyone. I can't find anywhere from spec whether it's signed or unsigned, I'd expect it to be read as unsigned int16 though. I wonder if some crazy generated code could hit this limit 🤔

Reply:
It is unsigned.

We have number of tests that use maxstack 65535, e.g. https://github.com/dotnet/runtime/blob/72e1ece7b683301a3ee9e0d7e8642535b4e6e89e/src/tests/JIT/Regression/JitBlue/DevDiv_359734/DevDiv_359734.il#L22

Reply:
Reverted in https://github.com/dotnet/runtime/pull/97794/commits/d6b2b334eb09f44f2e44e76b4157c1b003b9d98a